### PR TITLE
Create a release and tag

### LINKED ISSUE #79: Generate automatic release
Hi, @hectorerb.

Could you generate automatic release description from git metadata
https://github.com/conventional-changelog/conventional-github-releaser

The idea is to create a link with the reference to download the files.

Regards.

### PR DESCRIPTION
**Pull request by hectorerb**
This PR reorder travis to work with this work-flow:

- install jq for json parse in command-line
- separates scripts by types
- install standard-version scope global
- install conventional-github-releaser scope global
- execute fastlane test on develop and master branch
- build the code on features branch
- configure git
- generate CHANGELOG.md and increment version number
- create a tag with version number
- send to Google Play like a release

close: https://github.com/flyve-mdm/flyve-mdm-ios-agent/issues/79

### CODE REVIEW

File: ci/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        # Push tag to github
+        # conventional-github-releaser -t $GH_TOKEN -r 0
+        # Get version number from package.json
+        export GIT_TAG=$(jq -r ".version" package.json)
+        # Update CFBundleShortVersionString
+        /usr/libexec/PlistBuddy -c "Set :CFBundleShortVersionString ${GIT_TAG}" ${PWD}/${APPNAME}/Info.plist
+        # Add modified and delete files
+        git add -u
+        # Create commit
+        git commit -m "ci(build): increment **version** ${GIT_TAG}"
```
Comment (ajsb85) [inline ci/deploy.sh:50]: I don't like the bold (**version**) in the commit message  :P

File: ci/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+elif [[ "$TRAVIS_BRANCH" == "master" && "$TRAVIS_PULL_REQUEST" == "false" ]]; then
+    git config --global user.email $GH_EMAIL
+    git config --global user.name "Flyve MDM"
+    git remote remove origin
+    git remote add origin https://$GH_USER:$GH_TOKEN@github.com/flyve-mdm/flyve-mdm-ios-agent.git
+
+    if [[ $TRAVIS_COMMIT_MESSAGE != *"**version**"* && $TRAVIS_COMMIT_MESSAGE != *"**CHANGELOG.md**"* ]]; then
+        git checkout $TRAVIS_BRANCH -f
+        # Generate CHANGELOG.md and increment version
+        npm run release -- -t '' -m "ci(release): generate **CHANGELOG.md** for version %s"
```
Comment (ajsb85) [inline ci/deploy.sh:40]: I don't like the bold (**CHANGELOG.md**) in the commit message :P

File: ci/deploy.sh (hunk 1 above)
Comment (hectorerb) [reply ci/deploy.sh:50]: @ajsb85 me either, but it's to be able to identify this commit

File: ci/before_install.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+# @date      25/08/17
+# @copyright Copyright © 2017 Teclib. All rights reserved.
+# @license   GPLv3 https://www.gnu.org/licenses/gpl-3.0.html
+# @link      https://github.com/flyve-mdm/flyve-mdm-ios-agent
+# @link      https://.flyve-mdm.com
+# ------------------------------------------------------------------------------
+
+echo ----------- Create Fastlane environment variables ------------
+# Create Fastlane environment variables
+echo FASTLANE_PASSWORD=$FASTLANE_PASSWORD >> .env
```
Comment (ajsb85) [inline ci/before_install.sh]: Where you store the $FASTLANE_PASSWORD?
In an encrypted variable?

**Review by ajsb85 — APPROVED** [review]
Reviewed

File: ci/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+if [[ "$TRAVIS_BRANCH" == "develop" && "$TRAVIS_PULL_REQUEST" == "false" ]]; then
+  fastlane beta
+
+elif [[ "$TRAVIS_BRANCH" == "master" && "$TRAVIS_PULL_REQUEST" == "false" ]]; then
+    git config --global user.email $GH_EMAIL
+    git config --global user.name "Flyve MDM"
+    git remote remove origin
+    git remote add origin https://$GH_USER:$GH_TOKEN@github.com/flyve-mdm/flyve-mdm-ios-agent.git
+
+    if [[ $TRAVIS_COMMIT_MESSAGE != *"**version**"* && $TRAVIS_COMMIT_MESSAGE != *"**CHANGELOG.md**"* ]]; then
```
Comment (hectorerb) [inline ci/deploy.sh:37]: @ajsb85 Here I identify if it's that commit

File: ci/deploy.sh (hunk 2 above)
Comment (hectorerb) [reply ci/deploy.sh:40]: This commit is catched because CHANGELOG.md should not be created more than once

**Pull request merged**